overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt

```diff
@@ -42,39 +45,74 @@ class PanelsChildGestureRegionObserver : View.OnLayoutChangeListener {
     oldRight: Int,
     oldBottom: Int
   ) {
-    if (view != null) {
-      val coordinates = intArrayOf(0, 0)
-      view.getLocationOnScreen(coordinates)
-
-      val x = coordinates[0]
-      val y = coordinates[1]
-
-      val absoluteLeft = x + left
-      val absoluteTop = y + top
-      val absoluteRight = x + right
-      val absoluteBottom = y + bottom
-
-      viewIdToGestureRegionMap[view.id] = Rect(
-        absoluteLeft,
-        absoluteTop,
-        absoluteRight,
-        absoluteBottom
-      )
+    val coordinates = intArrayOf(0, 0)
+    view.getLocationInWindow(coordinates)
+
+    val x = coordinates[0]
+    val y = coordinates[1]
+
+    val absoluteRight = x + right
+    val absoluteBottom = y + bottom
 
-      publishGestureRegionsUpdate()
+    viewIdToGestureRegionMap[view.id] = Rect(
+      x,
+      y,
+      absoluteRight,
+      absoluteBottom
+    )
+
+    publishGestureRegionsUpdate()
+  }
+
+  @UiThread
+  fun register(view: View) {
+    view.addOnLayoutChangeListener(this)
+
+    val listener = ViewTreeObserver.OnScrollChangedListener {
+      onLayoutChange(
+        view = view,
+        left = view.left,
+        top = view.top,
+        right = view.right,
+        bottom = view.bottom,
+        oldLeft = 0,
+        oldTop = 0,
+        oldRight = 0,
+        oldBottom = 0
+      )
     }
+
+    view.viewTreeObserver.addOnScrollChangedListener(listener)
+    viewIdToListenerMap[view.id] = listener
   }
 
   /**
    * Stop publishing gesture region updates based on layout changes to android.view.View
    * corresponding to [viewId].
    */
+  @Deprecated(
+    message = "Use unregister instead",
+    replaceWith = ReplaceWith("unregister(view)")
+  )
   @UiThread
   fun remove(viewId: Int) {
     viewIdToGestureRegionMap.remove(viewId)
     publishGestureRegionsUpdate()
   }
 
+  /**
+   * Stop publishing gesture region updates based on layout and scroll changes to android.view.View
+   */
+  @UiThread
+  fun remove(view: View) {
+    viewIdToListenerMap.remove(view.id)?.let {
+      view.viewTreeObserver.removeOnScrollChangedListener(it)
+    }
```

Comment:
Previously, `addOnLayoutChangeListener` was called outside of this class. Now that we're calling it inside this class, should we also call `removeOnLayoutChangeListener` in this `remove` / `unregister` method?